### PR TITLE
#3234 Start at Home

### PR DESCRIPTION
**Pull request by electricRGB**
- Adds section in settings for Start at Home
- Start at Home now checks on app becoming active & cold launch
- adds telemetry for Start at Home setting

### CODE REVIEW

File: Client/Frontend/Browser/TabManager.swift [hunk 1]
```diff
@@ -704,6 +706,122 @@ extension TabManager {
...
+        if setting == .afterFourHours {
+            timeSinceLastActivity = dateComponents.hour ?? 0
+            timeToOpenNewHome = 4
+
+        } else if setting == .always {
+            timeSinceLastActivity = dateComponents.second ?? 0
+            timeToOpenNewHome = 5
+        }
+
+        return timeSinceLastActivity >= timeToOpenNewHome //|| coldLaunch
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Client/Frontend/Browser/TabManager.swift:766]: This is actually a mistake. Will fix.

File: Client/Frontend/Settings/HomePageSettingViewController.swift [hunk 2]
```diff
@@ -111,49 +111,52 @@ class HomePageSettingViewController: SettingsTableViewController, FeatureFlagsPr
...
+
+            let extras = [TelemetryWrapper.EventExtraKey.preference.rawValue: PrefsKeys.StartAtHome,
+                          TelemetryWrapper.EventExtraKey.preferenceChanged.rawValue: option.rawValue]
+            TelemetryWrapper.recordEvent(category: .action, method: .change, object: .setting, extras: extras)
+        }
+
+        let afterFourHoursOption = CheckmarkSetting(
+            title: NSAttributedString(string: .Settings.Homepage.StartAtHome.AfterFourHours),
+            subtitle: nil,
+            accessibilityIdentifier: "StartAtHomeAfterFourHours",
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Client/Frontend/Settings/HomePageSettingViewController.swift:130]: Great point. I totally forgot.

**Comment by nishant2718** [comment]
Does the `metrics.yaml` change require a data review?

File: Client/Application/AccessibilityIdentifiers.swift [hunk 3]
```diff
@@ -22,4 +22,14 @@ public struct AccessibilityIdentifiers {
...
+
+    struct Settings {
+        struct Homepage {
+            struct StartAtHome {
+                static let afterFourHours = "StartAtHomeAfterFourHours"
+                static let always = "StartAtHomeAlways"
+                static let disabled = "StartAtHomeDisabled"
+            }
+        }
+    }
```
Comment (nishant2718) [inline Client/Application/AccessibilityIdentifiers.swift:34]: 👌

**Review by nishant2718 — APPROVED** [review]

**Pull request merged**